Storage/Storage/Model/Customer+CoreDataProperties.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import CoreData
+
+
+extension Customer {
+
+    @nonobjc public class func fetchRequest() -> NSFetchRequest<Customer> {
+        return NSFetchRequest<Customer>(entityName: "Customer")
+    }
+
+    @NSManaged public var customerID: Int64
+    @NSManaged public var email: String?
+    @NSManaged public var firstName: String?
+    @NSManaged public var lastName: String?
+    @NSManaged public var billingFirstName: String?
+    @NSManaged public var billingLastName: String?
+    @NSManaged public var billingCompany: String?
+    @NSManaged public var billingAddress1: String?
+    @NSManaged public var billingAddress2: String?
+    @NSManaged public var billingCity: String?
+    @NSManaged public var billingState: String?
+    @NSManaged public var billingPostcode: String?
+    @NSManaged public var billingCountry: String?
+    @NSManaged public var billingPhone: String?
+    @NSManaged public var billingEmail: String?
+    @NSManaged public var shippingFirstName: String?
+    @NSManaged public var shippingLastName: String?
+    @NSManaged public var shippingCompany: String?
+    @NSManaged public var shippingAddress1: String?
+    @NSManaged public var shippingAddress2: String?
+    @NSManaged public var shippingCity: String?
+    @NSManaged public var shippingState: String?
+    @NSManaged public var shippingPostcode: String?
+    @NSManaged public var shippingCountry: String?
+    @NSManaged public var shippingPhone: String?
+    @NSManaged public var shippingEmail: String?
```

Comment:
The tests suggest these are non-optional, however they are shown as optional both here and in the inspector for `Model 74`. Should any of these be non-optional in the database?

Reply:
Gotcha! These should be optional, so I modified these in da814bf07cd4f2c97e6fe2a9a325ffa15585cbd6 , rather than testing for values in the data model, I'm testing now for property existence.